src/keycloak/keycloak_openid.py

```diff
@@ -370,11 +378,14 @@ def exchange_token(
         params_path = {"realm-name": self.realm_name}
         payload = {
             "grant_type": ["urn:ietf:params:oauth:grant-type:token-exchange"],
-            "client_id": client_id,
+            "client_id": self.client_id,
```

Comment:
I´m not sure I´m entirely on board with this change :) Removing `client_id` from the input arguments is a breaking change, we will be forcing users to rewrite their code with this change (including myself as I'm heavily dependent on this method). Which by itself is not a bad thing, but I want to have a clear justification for this change.

The reason why I'm not fully on board, is that the client id is tightly linked to the token, i.e. the token has to be generated by the same client. So if you pass a token not generated by the client, you don't really have a simple way to change that. On the other hand, I do understand it from the encapsulation concept of the KeycloakOIDClient class. But we are reducing the capabilities of this method now, it is not "just exchange any token to any token", now it is becoming "exchange current client's tokens only to other tokens".

Reply:
I completely understand and empathize with not wanting to introduce a breaking change. I can remove this change and we can discuss introducing it in a different pull-request.

This being said I still believe it's an important change in terms of the encapsulation of the KeycloakOIDClient class. If you want to have a method that allows to "just exchange any token to any token" I believe it should be a static function and the client_secret needs to be an argument or you'll run into problems like this the one in this issue  #380. In my opinion this method should be a "exchange current client's tokens only to other tokens" type of behavior, especially because in its implementation it gets the client secret from the classes attributes. (`payload = self._add_secret_key(payload)`)

For this reasons I believe it needs to be either a static method of the class, and both the client_id and client_secret need to be arguments of the static method, or an instance method that gets the client_id from the instance.

Either way, as I mentioned in the beginning of this reply, this is not the main subject of this pull request. My main goal is to be able to use the requested_issuer field to exchange my token for an IDP token. And so, if you want to move this to a different Pull Request I'm more than happy to make the necessary changes and submit this specific alteration in a different Pull Request.

Reply:
Nah I think you're right :) It makes more sense the way you propose. I think it's fine, let's just make sure the conventional commit message contains the BREAKING CHANGE indication :)

Reply:
Sounds good. I'm a bit busy but when I have a bit of free time I'll do that.
I'll need to change some of the unit tests that are failing relating to this method. Should I include that message in that particular commit or do I have to put in both?

Reply:
Only 1 is enough :)

Reply:
Sorry for the long time it took me to get back to this. I've been busy as hell but this was not forgotten! Happy to be able to contribute and I hope my contribution is well appreciated :smile: